app/controllers/api/v1/move_events_controller.rb

```diff
@@ -62,12 +68,25 @@ def validate_params
         MoveEvents::ParamsValidator.new(event_params).validate!
       end
 
+      def validate_relationships
```

Comment:
I don't think this is really validating relationships, but rather defining different strong params for each type of different event. I would suggest we pass in different `event_params` per action, this way we are explicit in which params we allow for different types of actions, and we don't tie ourselves down further? So we can define some shared params under `base_params` maybe and do something like this:
```
def lockout_params
 ...
end

def redirect_params
  ...
end

def create_event(event_name, event_params)
...
end
```

Reply:
interesting idea, I'll have a go with that